### PR TITLE
add missing state change code, add more unit test code

### PR DESCRIPTION
**Pull request by snazy**
* split utests in CassandraSchedulerTests into separate source files
* fix mesos-resource-role matchin (match everything on `*`)
* fix buggy cluster job state handling when executor is lost or task is lost
* add more utests + enhance existing utests (should increase code coverage)
* fix buggy change-seed code

### CODE REVIEW

File: cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java [hunk 1]
```diff
@@ -823,13 +827,13 @@ private static CassandraNodeTask getMetadataTask(@NotNull final String executorI
         final Long availableMem = resourceValueLong(headOption(index.get("mem"))).or(0L);
         final Long availableDisk = resourceValueLong(headOption(index.get("disk"))).or(0L);
 
-        if (availableCpus <= cpu) {
-            errors.add(String.format("Not enough cpu resources for role %s. Required %f only %f available.", mesosRole, cpu, availableCpus));
+        if (availableCpus < cpu) {
```
Comment (BenWhitehead) [inline cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java]: Why the change to strictly less than? Won't this now fail if the task needs `2` and the offer has exactly `2`?

File: cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java [hunk 2]
```diff
@@ -339,7 +339,7 @@ public ResourceHasExpectedRole(@NotNull final String role) {
 
         @Override
         public boolean apply(final Resource item) {
-            return item.getRole().equals(role);
+            return "*".equals(role) || "*".equals(item.getRole()) || item.getRole().equals(role);
```
Comment (BenWhitehead) [inline cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java]: I don't think we want to absorb the extra logic necessary to be able to combine offers right now to the point that we could use resources across multiple roles.  Please remove this for now, we can decide if we want to do something around multi-role support later on.

Comment (BenWhitehead) [inline cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java]: For simplicity this should be an all or nothing match.

Either all resources will be available in the offer matching the specified role, or not.

Also I think `"*".equals(role)` could cause true to be returned in a scenario where the resource didn't actually have the `*` role.

File: cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java (hunk 1 above)
Comment (snazy) [inline cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java]: Nope - the unit test unveiled that exactly this scenario failed (offer==X and requirement==X). (We only want to fail, if the offer has less resources)

**Comment by snazy** [comment]
Regarding roles: so people must provide a role name? Just wanted to make sure, that if there is no role management enabled, we don't accidentally reject all offers because `*` doesn't match.

File: cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java (hunk 1 above)
Comment (BenWhitehead) [inline cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java]: Makes sense, you're right. We're checking the negative in this case not the positive.

**Comment by BenWhitehead** [comment]
The existing code tries to read the environment variable and use the value that is provided by the user, otherwise it defaults to `*`. That means there is always a value defined and all later code has to match the specified role regardless of what it's set to.

Please remove the special null handling and just use the exact match for the value passed.

**Comment by snazy** [comment]
OK, removed that special `*` role handing exceptions.
Also rebased against `rewrite`
